tests/components/netatmo/test_climate.py

```diff
@@ -359,6 +365,174 @@ async def test_service_preset_modes_thermostat(
+            ATTR_TARGET_TEMPERATURE: 25,
+            ATTR_TIME_PERIOD: "02:24:00",
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # Test webhook room mode change to "manual"
+    response = {
+        "room_id": "2746182631",
+        "home": {
+            "id": "91763b24c43d3e344f424e8b",
+            "name": "MYHOME",
+            "country": "DE",
+            "rooms": [
+                {
+                    "id": "2746182631",
+                    "name": "Livingroom",
+                    "type": "livingroom",
+                    "therm_setpoint_mode": "manual",
+                    "therm_setpoint_temperature": 25,
+                    "therm_setpoint_end_time": 1612749189,
+                }
+            ],
+            "modules": [
+                {"id": "12:34:56:00:01:ae", "name": "Livingroom", "type": "NATherm1"}
+            ],
+        },
+        "mode": "manual",
+        "event_type": "set_point",
+        "push_type": "display_change",
+    }
+    await simulate_webhook(hass, webhook_id, response)
+
+    assert hass.states.get(climate_entity_livingroom).state == "heat"
+    assert hass.states.get(climate_entity_livingroom).attributes["temperature"] == 25
+
+
+async def test_service_clear_temperature_setting(
+    hass: HomeAssistant, config_entry, netatmo_auth
+) -> None:
+    """Test service clearing temperature setting."""
+    with selected_platforms(["climate"]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+
+        await hass.async_block_till_done()
+
+    webhook_id = config_entry.data[CONF_WEBHOOK_ID]
+    climate_entity_livingroom = "climate.livingroom"
+
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+
+    # Test service setting the temperature without an end datetime
+    await hass.services.async_call(
+        NETATMO_DOMAIN,
+        SERVICE_CLEAR_TEMPERATURE_SETTING,
+        {ATTR_ENTITY_ID: climate_entity_livingroom},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # Test webhook room mode change to "manual"
+    response = {
+        "room_id": "2746182631",
+        "home": {
+            "id": "91763b24c43d3e344f424e8b",
+            "name": "MYHOME",
+            "country": "DE",
+            "rooms": [
+                {
+                    "id": "2746182631",
+                    "name": "Livingroom",
+                    "type": "livingroom",
+                    "therm_setpoint_mode": "home",
+                    "therm_setpoint_end_time": 1612749189,
+                }
+            ],
+            "modules": [
+                {"id": "12:34:56:00:01:ae", "name": "Livingroom", "type": "NATherm1"}
+            ],
+        },
+        "mode": "home",
+        "event_type": "set_point",
+        "push_type": "display_change",
+    }
+    await simulate_webhook(hass, webhook_id, response)
+
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
```

Comment:
Does this really test anything as it already was auto?

Reply:
I'm a bit unsure about all of the tests here tbh. This is probably one for @cgtobi to comment on.

The process of the services is:

1. User makes service call
2. Call to API is made through external library
3. Netatmo sends a request to webhook
4. HA updates devices based on webhook

From what I can tell of the tests, only 1, 3 and 4 are being checked.

To compartmentalise the tests, I expect it would be better to have tests to check parts 1 and 2, then separate tests to check parts 3 and 4. I'm not familiar enough with the testing in HA though to know how to do that. It would also increase the scope of this PR by quite a lot to change this.

Reply:
I mean in the last test just add so it does
1. Make service call to set a temperature with an end datetime
2. Check result
3. Make service call (with new service `async_service_clear_temperature_setting()`) to clear the setting
4. Check result

On the current test we're only asserting the state which is `auto` before and after so actually nothing changes in the test.
It's only doing 3 and 4 right now.

Reply:
Ah I see what you mean. I'll update the test.